server/src/main/java/org/opensearch/node/remotestore/RemoteStorePinnedTimestampService.java

```diff
@@ -262,6 +276,10 @@ public static Tuple<Long, Set<Long>> getPinnedTimestamps() {
         return pinnedTimestampsSet;
     }
 
+    public static HashMap<String, List<Long>> getPinnedEntities() {
+        return pinnedEntityToTimestampsMap;
```

Comment:
Check if `RemoteStorePinnedTimestampService.fetchPinnedTimestamps()` can be used here.

Reply:
It can be , but we are caching the timestamps . So going with the current implementation .